circuit/src/jvmTest/kotlin/com/slack/circuit/OverlayTest.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2022 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.circuit
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.key
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberUpdatedState
+import app.cash.molecule.RecompositionClock
+import app.cash.molecule.moleculeFlow
+import app.cash.turbine.test
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.test.runTest
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class OverlayTest {
+  @Test
+  fun noOverlayMeansNullData() = runTest {
+    moleculeFlow(RecompositionClock.Immediate) {
+        val overlayHost = rememberOverlayHost()
+        overlayHost.currentOverlayData
+      }
+      .distinctUntilChanged()
+      .test { assertNull(awaitItem()) }
+  }
+
+  @Test
+  fun overlayHasData() = runTest {
+    moleculeFlow(RecompositionClock.Immediate) {
+        val overlayHost = rememberOverlayHost()
+        LaunchedEffect(overlayHost) {
+          overlayHost.show(
+            object : Overlay<String> {
+              @Composable override fun Content(navigator: OverlayNavigator<String>) {}
+            }
```

Comment:
Why can't we make `Overlay` a `fun interface` and replace this with a lambda?

Reply:
hadn't occurred to me, not against it 👍

Reply:
urg, looks like this has the same anonymous function issues we saw with Presenter and Ui

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/1361086/194678421-7cbb63c0-2742-447b-910a-008aaf0e22c9.png">

Reply:
Sorry, my initial post was poorly worded - it was meant to be a question 😲. Thanks for trying